### ISSUE TITLE
batting_stats_range IndexError: list index out of range

### ISSUE THREAD

**Issue by sportsstats** [jldbc/pybaseball#74]
Receiving the ERROR below CODE below when trying to get batting stats based on date range. Can anyone provide any help with this?

**CODE**
from pybaseball import batting_stats_range
from pybaseball import pitching_stats_range

data = batting_stats_range('2017-05-01', '2017-05-08')
data.head()

**ERROR**
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-4-f74839c81f0c> in <module>
----> 1 data = batting_stats_range('2017-05-01', '2017-05-08')
      2 data.head()

~/opt/anaconda3/lib/python3.7/site-packages/pybaseball/league_batting_stats.py in batting_stats_range(start_dt, end_dt)
     79     # retrieve html from baseball reference
     80     soup = get_soup(start_dt, end_dt)
---> 81     table = get_table(soup)
     82     table = table.dropna(how='all')  # drop if all columns are NA
     83     # scraped data is initially in string format.

~/opt/anaconda3/lib/python3.7/site-packages/pybaseball/league_batting_stats.py in get_table(soup)
     49 
     50 def get_table(soup):
---> 51     table = soup.find_all('table')[0]
     52     data = []
     53     headings = [th.get_text() for th in table.find("tr").find_all("th")][1:]

IndexError: list index out of range

**Comment by parkerwardjohnson** [comment]
Getting the same error and haven't been able to troubleshoot. Anyone's help with this would be appreciated.

**Comment by nickwan** [comment]
same

**Comment by nickwan** [comment]
so this looks less like a `pybaseball` thing and more like a bbref thing. their daily game log finder seems to be "down" and looks like it'll be moving over behind a paywall at statshead. thought i had a fix for yall but turns out no queries work. might have to rework this entire function and use fangraphs or something instead... stay tuned

**Comment by blacktj** [comment]
When I navigate to the page: 

'http://www.baseball-reference.com/leagues/daily.cgi?user_team=&bust_cache=&type=b&lastndays=7&dates=fromandto&fromandto=2017-05-01.2017-05-08&level=mlb&franch=&stat=&stat_value=0'

You see this:

The Play Index will soon be shutting down. But don't despair!
We've launched new and improved offerings at Stathead! All Play Index subscriptions have been converted, or you can sign up for a free trial here!
Please read our blog post: Ad-Free and Play Index Changes Coming to Baseball-Reference.com.

**Comment by nickwan** [comment]
@blacktj yep, so this is in the `get_soup()` function in the `league_batting_stats.py` file. you can try changing the code from:
`url = "http://www.baseball-reference.com/leagues/daily.cgi?user_team=&bust_cache=&type=b&lastndays=7&dates=fromandto&fromandto={}.{}&level=mlb&franch=&stat=&stat_value=0".format(start_dt, end_dt)`

to 
`url = f"https://www.baseball-reference.com/leagues/daily.fcgi?request=1&type=b&lastndays=7&dates=since&since={start_dt}&fromandto={start_dt}.{end_dt}&level=mlb&franch=ANY"` (this is python 3.6+, with f-strings) 

but that doesn't fix anything because of bbref. currently working on a fangraphs solution:
https://www.fangraphs.com/leaders.aspx?pos=all&stats=bat&lg=all&qual=y&type=0&season=2017&month=1000&season1=2017&ind=0&team=0&rost=0&age=0&filter=&players=0&startdate=2017-05-01&enddate=2017-05-08&page=1_50

seems like you could rip like this but similar to bbref, if there's a bunch of nerds souping their data and no one is paying (and no one is visiting and having ads displayed) then this functionality will probably go behind a paywall too. i'd say if you can, please try supporting fangraphs before you use this workaround.

**Comment by nickwan** [comment]
k gonna make a PR for this... but if you are impatient, here's what you'll want to change: 

```
def get_soup(start_dt, end_dt):
    # get most recent standings if date not specified
    # if((start_dt is None) or (end_dt is None)):
    #    print('Error: a date range needs to be specified')
    #    return None
    url = f"https://www.fangraphs.com/leaders.aspx?pos=all&stats=bat&lg=all&qual=y&type=c,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,34,35,36,37,38,39,40,60,41,201,205,200,52,51,50,61,62,63,64,65,66,67,68,69,70,71,53,111,54,56,203,199,204,55,57,58,59&season={start_dt[:4]}&month=1000&season1={end_dt[:4]}&ind=0&startdate={start_dt}&enddate={end_dt}&page=1_1500"
    s = requests.get(url).content
    return BeautifulSoup(s, "lxml")

def get_table(soup):
    table = soup.find_all('table')[16]
    data = []
    headings = [th.get_text() for th in table.find_all("th")]
    data.append(headings)
    table_body = table.find('tbody')
    rows = table_body.find_all('tr')
    for row in rows:
        cols = row.find_all('td')
        cols = [ele.text.strip() for ele in cols]
        data.append([ele for ele in cols])
    data = pd.DataFrame(data)
    data = data.rename(columns=data.iloc[0])
    data = data.reindex(data.index.drop(0))
    return data
``` 

i hard coded all the stats from the standard, advanced, win probability, and value tabs. 
if someone wants to review the PR that'd be sweet -- and if @jldbc wants to approve it in a sec that'd rule

**Comment by nickwan** [comment]
alright fam, PR is up 
https://github.com/jldbc/pybaseball/pull/76

**Comment by nickwan** [comment]
one more PR for review and seemingly immediate merge, except for the pitching side 
https://github.com/jldbc/pybaseball/pull/77

**Comment by schorrm** [comment]
@nickwan make the PR on my fork and I'll work on integrating it

**Comment by schorrm** [comment]
Seems to be fixed with #82 - merging the fork in
